aten/src/ATen/native/Gesv.cpp

```diff
@@ -72,6 +78,100 @@ static void applyGesv(Tensor& b, Tensor& A, std::vector<int64_t> infos) {
   }
 }
 
+std::tuple<Tensor&,Tensor&> _gesv_single_out_cpu(
+    Tensor& sol, Tensor& lu,
+    const Tensor& self, const Tensor& A) {
+#ifndef USE_LAPACK
+  AT_ERROR("gesv: LAPACK library not found in compilation");
+#endif
+  int64_t bx = self.size(0);
+  int64_t by = (self.dim() == 1) ? 1 : self.size(1);
+  int64_t ax = A.size(0);
+  int64_t ay = A.size(1);
+  int info = 0;
+
+  bool use_temp_sol = false;
+  bool use_temp_lu = false;
+  auto temp_sol = self.type().tensor();
+  auto temp_lu = self.type().tensor();
+
+  /* Init to column major format. `sol` and `lu` need to be contiguous
+   * since we pass sol.data() and lu.data() to Lapack */
+
+  bool tc = isTransposeContiguous(sol);
+  if (tc) {
+    // if transpose of output tensor is contiguous, use it
+    sol.t_();
```

Comment:
I'm worried about this. It makes the function not re-entrant anymore. If an error is thrown before the final `sol_` near the end of this function, it is transposed forever! This can very well happen if input is half type or integral types because of the `AT_DISPATCH_FLOATING_TYPES` below. Can we not use the inplace version?

Reply:
Switched to using sol = sol.t(). This still reassigns the reference though

Reply:
moved the AT_ERROR out of AT_DISPATCH_FLOATING_TYPES to the end, after sol and lu have been transposed again. Should fix the re-entrant safeness